File: src/main/java/net/chcknlvr/blockx/item/TungstenMaterial.java
package net.chcknlvr.blockx.item;

import net.chcknlvr.blockx.Blockx;
import net.minecraft.item.ToolMaterial;
import net.minecraft.recipe.Ingredient;

public class TungstenMaterial implements ToolMaterial {
    public static final TungstenMaterial INSTANCE = new TungstenMaterial();
    @Override
    public int getDurability() {
        return 5000;
    }

    @Override
    public float getMiningSpeedMultiplier() {
        return 20;
    }

    @Override
    public float getAttackDamage() {
        return 10;
    }

    @Override
    public int getMiningLevel() {
        return 6;
    }

    @Override
    public int getEnchantability() {
        return 50;
    }

    @Override
    public Ingredient getRepairIngredient() {
        return Ingredient.ofItems(Items.TUNGSTEN_INGOT);
    }
}

File: src/main/java/net/chcknlvr/blockx/item/TungstenMaterial.kt
package net.chcknlvr.blockx.item

import net.minecraft.item.ToolMaterial
import net.minecraft.recipe.Ingredient

class TungstenMaterial : ToolMaterial {
    override fun getDurability(): Int {
        return 5000
    }

    override fun getMiningSpeedMultiplier(): Float {
        return 20.0f
    }

    override fun getAttackDamage(): Float {
        return 10.0f
    }

    override fun getMiningLevel(): Int {
        return 5
    }

    override fun getEnchantability(): Int {
        return 30
    }

    override fun getRepairIngredient(): Ingredient {
        return Ingredient.ofItems(Items.TUNGSTEN_INGOT)
    }

    companion object {
        @JvmField
        val INSTANCE = TungstenMaterial()
    }
}